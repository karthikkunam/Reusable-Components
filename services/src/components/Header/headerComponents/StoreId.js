import React from 'react';
import PropTypes from 'prop-types';

function StoreId(props) {
  const { storeID } = props;
  return <div>{storeID && <span className="navbar-text header-text">Store: {storeID}</span>}</div>;
}
StoreId.propTypes = {
  storeID: PropTypes.string,
};

StoreId.defaulProps = {
  storeID: '',
};

export default StoreId;
