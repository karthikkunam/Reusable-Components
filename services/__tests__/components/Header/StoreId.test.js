import React from 'react';
import StoreId from '../../../src/components/Header/headerComponents/StoreId';
import { shallow } from 'enzyme';

describe('StoreId Component', () => {
  it('renders correctly', () => {
    const props = {
      storeID: 35678,
    };
    const component = shallow(<StoreId {...props} />);
    expect(component).toMatchSnapshot();
  });
  it('test without storeID', () => {
    const props = {
      storeID: undefined,
    };
    const component = shallow(<StoreId {...props} />);
    expect(component).toMatchSnapshot();
  });
});
