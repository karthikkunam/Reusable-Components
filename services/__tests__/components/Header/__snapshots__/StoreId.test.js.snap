// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreId Component renders correctly 1`] = `
<div>
  <span
    className="navbar-text header-text"
  >
    Store: 
    35678
  </span>
</div>
`;

exports[`StoreId Component test without storeID 1`] = `<div />`;
